### ISSUE TITLE
Browser crash upon consecutive seeds, possible memory leak? 

### ISSUE THREAD

**Issue by kbbqiu** [gitsummore/nile.js#26]
As we rack up the number WebTorrent seeds on the broadcaster side, there seems to be a client side memory leak which eventually causes the browser to crash. We’ve checked the CPU and memory usage but have found no error messages or anomalies.

Unsuccessful approaches that we have taken to solve the issue:
- Server side seeding in place of client side seeding
- Seeding in web workers
- Seeding in iframes
- Destroying WebTorrent client on each new seed
- Destroying WebTorrent file on each new seed

One **solution** that we have found, is browser specific. If using **Chrome Canary** to test the library, it seems to run indefinitely. If anyone in the open source community is able to provide some insight, please make a pull request on our GitHub repo.

**Comment by imvetri** [comment]
What are the steps to reproduce the issue in a local environment?

**Comment by no1youknowz** [comment]
Not sure I have the wrong end of the stick or not.

But wouldn't it be better to have another torrent server for the creating and deleting of the torrent files?

Perhaps using something like webtorrent-hybrid ?  

I'm sure this would off-load the requirement from the browser and better on a server?

**Comment by kbbqiu** [comment]
@imvetri we haven't updated the styling for the demo since we started the project so don't mind the css haha. Here's how to reproduce the problem. Using google chrome in this instance.

1. npm start
2. Got to localhost://8000/
3. Open a broadcaster screen
4. Open a viewer screen as well
5. Start the broadcaster
6. Viewer should begin in about 10 seconds or so
7. After some time, your browser should freeze...

**Comment by kbbqiu** [comment]
@no1youknowz same issue persists. We've tried using webtorrent hybrid on the server end. We still get the same issue, after about 10 emits, there is the warning saying that event emitters exceeded 10 and possible memory leak. From webTorrent github closed issues, supposedly the event emitter error isn't a big deal. 

The reason why we chose to go with client side seeding is so the server does less work. In Nile.js, the server only propogates the magnet URI to the rest of the network.

**Comment by no1youknowz** [comment]
@kbbqiu many thanks for the follow up.

**Comment by imvetri** [comment]
@kbbqiu can you share which node/ npm I should try this with. Currently tried npm install with following versions. 
Node - 7.7.3
Npm - 4.1.2

Ended with following error in the log file. It would be great if you confirm if this is not from nile.js package, I suspect issue with node/ npm. based on your confirmation I will open a issue there.

[npm-debug.log.txt](https://github.com/gitsummore/nile.js/files/1039024/npm-debug.log.txt)

**Comment by imvetri** [comment]
I tried with different version of node and npm. I got different error now.

FYI, I am running this in windows 7

node v6.10.1
npm  v3.10.10

Attaching debu log
[npm-debug.log.node6.txt](https://github.com/gitsummore/nile.js/files/1039074/npm-debug.log.node6.txt)

**Comment by derekmiranda** [comment]
Hi, @imvetri, seems like there's an issue with the postinstall, which is there to fix this [issue we had experienced with socket.io](https://github.com/socketio/socket.io/issues/1606). we personally have not tested this on windows systems but we'll look into it

**Comment by imvetri** [comment]
haha. Thanks for the update. 

Moreover I tried npm install on debugger.html project as well. It isn't working for that project too. Raised an issue there as well. https://github.com/devtools-html/debugger.html/issues/3045

I am suspecting its most likely issue with npm, if so I will add this in their bugs.

**Comment by imvetri** [comment]
Need some suggestions to mock the video stream. I do not have a camera on my desktop.

**Comment by imvetri** [comment]
@kbbqiu I tried the steps. What I observed is that after a while, viewer gets hung up, like the video will be stuck. Can you confirm whether this is the reported bug ?

**Comment by Nastybastard** [comment]
blockquote, div.yahoo_quoted { margin-left: 0 !important; border-left:1px #715FFA solid !important; padding-left:1ex !important; background-color:white !important; } Yes and I'm also having trouble with my battery changing and loading apps


Sent from Yahoo Mail for iPhone


On Thursday, June 8, 2017, 1:14 AM, Vetrivel <notifications@github.com> wrote:


@kbbqiu I tried the steps. What I observed is that after a while, viewer gets hung up, like the video will be stuck. Can you confirm whether this is the reported bug ?

**Comment by kbbqiu** [comment]
@imvetri yep. That should be the issue. There are no errors, just suddenly gets hung up.

**Comment by idkjs** [comment]
I guess this isnt worked out yet? Any updates?

**Comment by ChrisChiasson** [comment]
Is this issue still present in the latest chrome?
